### ISSUE TITLE
Segmentation fault (mri_vol2vol.bin)

### ISSUE THREAD

**Issue by kellyuw** [poldracklab/fmriprep#1681]
Hi all,

I'm trying to process a dataset with 30 subjects. Each subject has ~10 separate scan sessions (consisting of one structural / T1 scan and one resting-state scan). 

I've been running fmriprep-1.4.1rc4 with the following command: 
`singularity run --cleanenv -B ${LAB_DIR},/tmp ${CONTAINER} ${BIDS_DIR} ${WORK_DIR} participant --participant-label ${PARTICIPANT} --work-dir ${WORK_DIR} -vv --output-spaces PNC MNI152NLin2009cAsym:res-2 T1w fsnative --fd-spike-threshold 0.5 --notrack --return-all-components --stop-on-first-crash --ignore fieldmaps slicetiming sbref --fs-license-file ${BIDS_DIR}/derivatives/license.txt
`

I've run into the following error ([full log here: crash-logfile.txt](https://github.com/poldracklab/fmriprep/files/3336342/crash-logfile.txt)) and am unclear about how I might be able to resolve the issue:

```
mri_vol2vol --interp nearest --lta /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_01_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta --mov /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_01_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_bold_corrected_trans_masked.nii.gz --targ /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/freesurfer/sub-1018/mri/brainmask.mgz --o /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_01_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_bold_corrected_trans_masked_warped.nii.gz
Standard output:

Standard error:
/opt/freesurfer/bin/mri_vol2vol: line 3: 45044 Segmentation fault      (core dumped) mri_vol2vol.bin "$@"
Return code: 139
```

I found a [post on the FreeSurfer message board about long file paths causing a systematic Segmentation fault]( https://mail.nmr.mgh.harvard.edu/pipermail//freesurfer/2018-September/058546.html), which I think may be relevant. It looks like there have also been efforts to address this issue with fmriprep (as described in #1268 ). 

I also have found that the registration.lta files from different scan sessions for the same subject are pretty different from each other. The file path included in the registration.lta file is the same length for both sessions, so I'm unclear about why it only seems to be too long in the first registration.lta file?

**func_preproc_ses_05_task_rest_wf registration.lta**
```
# transform file /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_05_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta
# created by UNKNOWN on Thu Jun 27 10:20:50 2019

type      = 0 # LINEAR_VOX_TO_VOX
nxforms   = 1
mean      = 0.0000 0.0000 0.0000
sigma     = 10000.0000
1 4 4
2.799596309661865e+00 -3.733774647116661e-02 3.784747421741486e-02 2.017866897583008e+01
2.886015549302101e-02 -4.318719729781151e-02 -3.599380731582642e+00 1.708274230957031e+02
3.778534010052681e-02 2.799418210983276e+00 -5.502362921833992e-02 2.232846260070801e+01
0.000000000000000e+00 0.000000000000000e+00 0.000000000000000e+00 1.000000000000000e+00
src volume info
valid = 1  # volume info valid
filename = /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_05_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/uni_xform_masked.nii.gz
volume = 80 80 37
voxelsize = 2.799999952316284e+00 2.799999952316284e+00 3.599999904632568e+00
xras   = -9.960659146308899e-01 -7.030539959669113e-02 5.394337326288223e-02
yras   = -6.533798575401306e-02 9.938957095146179e-01 8.889498561620712e-02
zras   = 5.986388772726059e-02 -8.502071350812912e-02 9.945791959762573e-01
cras   = -4.082206726074219e+00 1.741598510742188e+01 3.047521591186523e+01
dst volume info
valid = 1  # volume info valid
filename = /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/freesurfer/sub-1018/mri/brainmask.mgz
volume = 256 256 256
voxelsize = 1.000000000000000e+00 9.999999403953552e-01 1.000000000000000e+00
xras   = -1.000000000000000e+00 2.168235369026661e-09 0.000000000000000e+00
yras   = 0.000000000000000e+00 -4.656613428188905e-10 -1.000000000000000e+00
zras   = -2.168235369026661e-09 1.000000000000000e+00 0.000000000000000e+00
cras   = -2.901954889297485e+00 1.142297363281250e+01 -2.492158508300781e+01
subject sub-1018
```

**func_preproc_ses_01_task_rest_wf registration.lta**
```
# transform file /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_01_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta# created by UNKNOWN on Wed Jun 26 09:45:51 2019type      = 0 # LINEAR_VOX_TO_VOXnxforms   = 1mean      = 0.0000 0.0000 0.0000sigma     = 10000.00001 4 42.799579620361328e+00 -1.558559108525515e-02 5.906276404857635e-02 1.863708496093750e+01 4.493862017989159e-02 -1.652508825063705e-01 -3.593260526657104e+00 1.771353149414062e+02 1.826756075024605e-02 2.795076131820679e+00 -2.121118754148483e-01 2.407226562500000e+01 0.000000000000000e+00 0.000000000000000e+00 0.000000000000000e+00 9.999999403953552e-01 src volume infovalid = 1  # volume info validfilename = path_too_long
volume = 80 80 37voxelsize = 2.799999952316284e+00 2.799999952316284e+00 3.599999904632568e+00xras   = -9.999048113822937e-01 1.371600478887558e-02 -1.485431450419128e-03yras   = 1.370038837194443e-02 9.998555779457092e-01 1.005676947534084e-02zras   = -1.623155549168587e-03 -1.003546174615622e-02 9.999483227729797e-01cras   = -3.232597351074219e+00 1.931093597412109e+01 -1.805091857910156e+00dst volume infovalid = 1  # volume info validfilename = /XXX/XXXXXXXXX/XXX/XXXXXX/XXXXXXXXXX/XXXXXXXXXX/XXXXXXXXXXXX/XXXXXXXXXXXXXXXXXXXX/derivatives/fmriprep-1.4.1rc4/freesurfer/sub-1018/mri/brainmask.mgzvolume = 256 256 256voxelsize = 1.000000000000000e+00 9.999999403953552e-01 1.000000000000000e+00xras   = -1.000000000000000e+00 2.168235369026661e-09 0.000000000000000e+00yras   = 0.000000000000000e+00 -4.656613428188905e-10 -1.000000000000000e+00zras   = -2.168235369026661e-09 1.000000000000000e+00 0.000000000000000e+00cras   = -2.901954889297485e+00 1.142297363281250e+01 -2.492158508300781e+01subject sub-1018
```

Thank you in advance for your time and consideration of this issue. Any advice about how to proceed would be very much appreciated!

**Comment by effigies** [comment]
Hi, thanks for the report. Would you be able to reformat this somewhat? It is fairly difficult to read, as the quote-text sometimes seems like it should be pre-formatted, and other times it's less clear. Could you use triple back-ticks to quote file contents? e.g.:

<pre>
```
type = 0 # LINEAR_VOX_TO_VOX
nxforms = 1
mean = 0.0000 0.0000 0.0000
sigma = 10000.0000
1 4 4
2.799145221710205e+00 -6.855968385934830e-02 1.176757644861937e-02 2.088180923461914e+01
7.354386616498232e-03 -7.339913398027420e-02 -3.598750591278076e+00 1.739473114013672e+02
6.877578049898148e-02 2.798197984695435e+00 -9.410997480154037e-02 1.840174293518066e+01
0.000000000000000e+00 0.000000000000000e+00 0.000000000000000e+00 1.000000119209290e+00
src volume info
valid = 1 # volume info valid
filename = BIDS_DIR/derivatives/fmriprep-1.4.1rc4/fmriprep_wf/single_subject_1018_wf/func_preproc_ses_06_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/uni_xform_masked.nii.gz
volume = 80 80 37
voxelsize = 2.799999952316284e+00 2.799999952316284e+00 3.599999904632568e+00
xras = -9.951207637786865e-01 -9.112911671400070e-02 3.781665116548538e-02
yras = -8.884717524051666e-02 9.943447709083557e-01 5.817771330475807e-02
zras = 4.290447011590004e-02 -5.453395098447800e-02 9.975897073745728e-01
cras = -4.955322265625000e+00 1.583534240722656e+01 2.515605163574219e+01
dst volume info
valid = 1 # volume info valid
filename = BIDS_DIR/derivatives/fmriprep-1.4.1rc4/freesurfer/sub-1018/mri/brainmask.mgz
volume = 256 256 256
voxelsize = 1.000000000000000e+00 9.999999403953552e-01 1.000000000000000e+00
xras = -1.000000000000000e+00 2.168235369026661e-09 0.000000000000000e+00
yras = 0.000000000000000e+00 -4.656613428188905e-10 -1.000000000000000e+00
zras = -2.168235369026661e-09 1.000000000000000e+00 0.000000000000000e+00
cras = -2.901954889297485e+00 1.142297363281250e+01 -2.492158508300781e+01
subject sub-1018
```
</pre>

And make clear when you've made substitutions. I suspect, for instance `BIDS_DIR` is not the name of the directory. When there are substitutions, please replace any censored characters with an `X` or similar, so that the number of characters is unchanged.

The reason I'm requesting this is that there are a couple reasons for a segfault in mri_vol2vol. While one is too-long filenames, which we try to guard against, another could be a lack of memory, and identifying whether you have too-long filenames will help determine whether our guards are failing or whether we should look elsewhere.

**Comment by kellyuw** [comment]
Awesome, thanks for the help with this, @effigies! I will edit my original post to use triple back-ticks when quoting file contents. I've also attached the raw lta files here, in case that's easier to read:
[raw-func_preproc_ses_05_task_rest_wf-bold_reg_wf-bbreg_wf-mri_coreg-registration.lta.txt](https://github.com/poldracklab/fmriprep/files/3346605/raw-func_preproc_ses_05_task_rest_wf-bold_reg_wf-bbreg_wf-mri_coreg-registration.lta.txt)
[raw-func_preproc_ses_01_task_rest_wf-bold_reg_wf-bbreg_wf-mri_coreg-registration.lta.txt](https://github.com/poldracklab/fmriprep/files/3346606/raw-func_preproc_ses_01_task_rest_wf-bold_reg_wf-bbreg_wf-mri_coreg-registration.lta.txt)

**Comment by effigies** [comment]
If ses_01 is unmodified, then it looks like the problem is that there are no new lines in the file... Can you try deleting the folder that it's found in, to let it re-run, and see if the problem persists?

**Comment by kellyuw** [comment]
Yes, it is unmodified, so I'll try deleting the folder and rerunning now. Thanks!

**Comment by soichih** [comment]
I am seeing the same error message - also running through the singularity.

```
singularity run --cleanenv docker://poldracklab/fmriprep:1.3.2 --notrack --resource-monitor --skip_bids_validation --stop-on-first-crash --use-plugin=multi_proc.yml --output-space fsaverage5 fsnative T1w template --template-resampling-grid=2mm --force-bbr --force-syn --skull-strip-template=NKI --fs-license-file=.license.txt --work-dir=fmripworkdir/ --participant_label=sub-TTTEMPSUB fmripinput/ fmripout/ participant
```

[stdout]
```
	 [Node] Running "mri_coreg" ("niworkflows.interfaces.freesurfer.PatchedMRICoregRPT"), a CommandLine Interface with command:
mri_coreg --s sub-TTTEMPSUB --dof 6 --ftol 1.000000e-04 --linmintol 1.000000e-02 --threads 7 --lta /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta --sep 4 --mov /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz --sd /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer
190719-17:23:14,166 nipype.interface INFO:
	 stderr 2019-07-19T17:23:14.165845:/opt/freesurfer/bin/mri_vol2vol: line 3:  5972 Segmentation fault      mri_vol2vol.bin "$@"
190719-17:23:14,246 nipype.workflow WARNING:
	 [Node] Error on "fmriprep_wf.single_subject_TTTEMPSUB_wf.func_preproc_ses_SSSES_task_rest_wf.bold_reg_wf.bbreg_wf.mri_coreg" (/N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg)
190719-17:23:15,355 nipype.workflow ERROR:
	 Node mri_coreg failed to run on host c45.
190719-17:23:15,360 nipype.workflow ERROR:
	 Saving crash info to /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/fmriprep/sub-TTTEMPSUB/log/20190719-151637_4e8539c5-f479-4d0f-b7ae-a8e19a409b1c/crash-20190719-172315-brlife-mri_coreg-b41ec0cd-a86f-44fd-8941-e3edb3d03dbd.txt
Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/legacymultiproc.py", line 69, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 473, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 557, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 637, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/freesurfer/base.py", line 144, in run
    return super(FSCommand, self).run(**inputs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 376, in run
    runtime = self._post_run_hook(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/freesurfer.py", line 222, in _post_run_hook
    runtime = super(TruncateLTA, self)._post_run_hook(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/registration.py", line 278, in _post_run_hook
    res = mri_vol2vol.run()
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/freesurfer/base.py", line 144, in run
    return super(FSCommand, self).run(**inputs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 375, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 758, in _run_interface
    self.raise_exception(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 695, in raise_exception
    ).format(**runtime.dictcopy()))
RuntimeError: Command:
mri_vol2vol --interp nearest --lta /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta --mov /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz --targ /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz --o /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked_warped.nii.gz
Standard output:

Standard error:
/opt/freesurfer/bin/mri_vol2vol: line 3:  5972 Segmentation fault      mri_vol2vol.bin "$@"
Return code: 139


```

[stderr]
```
...
Preprocessing did not finish successfully. Errors occurred while processing data from participants: TTTEMPSUB (1). Check the HTML reports for details.
Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/run.py", line 436, in main
    fmriprep_wf.run(**plugin_settings)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/workflows.py", line 599, in run
    runner.run(execgraph, updatehash=updatehash, config=self.config)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/base.py", line 161, in run
    self._clean_queue(jobid, graph, result=result))
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/base.py", line 224, in _clean_queue
    raise RuntimeError("".join(result['traceback']))
RuntimeError: Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/plugins/legacymultiproc.py", line 69, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 473, in run
    result = self._run_interface(execute=True)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 557, in _run_interface
    return self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/pipeline/engine/nodes.py", line 637, in _run_command
    result = self._interface.run(cwd=outdir)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/freesurfer/base.py", line 144, in run
    return super(FSCommand, self).run(**inputs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 376, in run
    runtime = self._post_run_hook(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/freesurfer.py", line 222, in _post_run_hook
    runtime = super(TruncateLTA, self)._post_run_hook(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/niworkflows/interfaces/registration.py", line 278, in _post_run_hook
    res = mri_vol2vol.run()
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/freesurfer/base.py", line 144, in run
    return super(FSCommand, self).run(**inputs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 375, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 758, in _run_interface
    self.raise_exception(runtime)
  File "/usr/local/miniconda/lib/python3.7/site-packages/nipype/interfaces/base/core.py", line 695, in raise_exception
    ).format(**runtime.dictcopy()))
RuntimeError: Command:
mri_vol2vol --interp nearest --lta /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta --mov /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz --targ /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz --o /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked_warped.nii.gz
Standard output:

Standard error:
/opt/freesurfer/bin/mri_vol2vol: line 3:  5972 Segmentation fault      mri_vol2vol.bin "$@"
Return code: 139


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/miniconda/bin/fmriprep", line 11, in <module>
    load_entry_point('fmriprep==1.3.2', 'console_scripts', 'fmriprep')()
  File "/usr/local/miniconda/lib/python3.7/site-packages/fmriprep/cli/run.py", line 440, in main
    sentry_sdk.capture_exception(e)
AttributeError: 'NoneType' object has no attribute 'capture_exception'
+ exitStatus=1
```

**Comment by effigies** [comment]
Hi @soichih can you post the contents of `/N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta`?

Another potential problem is that your actual filenames are longer than 255 characters.

```
$ for ARG in mri_vol2vol --interp nearest --lta /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta --mov /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz --targ /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz --o /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked_warped.nii.gz; do echo $( echo -n -- $ARG | wc -c ): $ARG; done
14: mri_vol2vol
11: --interp
10: nearest
8: --lta
231: /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta
8: --mov
260: /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz
9: --targ
146: /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz
6: --o
267: /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked_warped.nii.gz
```

`mri_vol2vol` has issues when the filename in an LTA file exceeded 255 characters; it's possible that they also have problems with actual filenames. Assuming that the problem isn't that your LTA file is missing newlines, as above, you could test this by copying your inputs into a local directory:

```Bash
$ cp /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_rest_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_image_valid_corrected_trans_masked.nii.gz /N/dc2/scratch/brlife/carbonate-workflows/5d29cd0f78d9fa1cee2a2474/5d31d9f378d9fa1cee2baea5/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz .
$ mri_vol2vol --interp nearest --lta registration.lta --mov ref_image_valid_corrected_trans_masked.nii.gz --targ brainmask.mgz --o ref_image_valid_corrected_trans_masked_warped.nii.gz
```

If that's the case, we'll need to come up with a workaround to ensure the inputs have shorter names by the time they get to `mri_vol2vol`.

**Comment by soichih** [comment]
@effigies 

Here is the command line we are endup running
```
mri_vol2vol --interp nearest 
--lta
/N/dc2/scratch/davhunt/workflows/5d6705bee0ae0618d1ededa8/5d673383e0ae06c2e9edef95/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_balloonanalogrisktask_run_1_wf/bold_reg_wf/bbreg_wf/mri_coreg/registration.lta

--mov 
/N/dc2/scratch/davhunt/workflows/5d6705bee0ae0618d1ededa8/5d673383e0ae06c2e9edef95/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_balloonanalogrisktask_run_1_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_bold_corrected_trans_masked.nii.gz

--targ 
/N/dc2/scratch/davhunt/workflows/5d6705bee0ae0618d1ededa8/5d673383e0ae06c2e9edef95/fmripout/freesurfer/sub-TTTEMPSUB/mri/brainmask.mgz

--o 
/N/dc2/scratch/davhunt/workflows/5d6705bee0ae0618d1ededa8/5d673383e0ae06c2e9edef95/fmripworkdir/fmriprep_wf/single_subject_TTTEMPSUB_wf/func_preproc_ses_SSSES_task_balloonanalogrisktask_run_1_wf/bold_reg_wf/bbreg_wf/mri_coreg/ref_bold_corrected_trans_masked_warped.nii.gz
```

**Comment by davhunt** [comment]
Hi @effigies thanks for your help, it does actually look like the problem is the registration.lta not having newlines--when I manually added them in it worked fine even with >255 character filenames.

**Comment by davhunt** [comment]
@effigies @soichih So the issue I believe is here in TruncateLTA mixin in freesurfer.py https://github.com/poldracklab/niworkflows/blob/master/niworkflows/interfaces/freesurfer.py#L217 , where a newline is appended for when the filename is too long but not in this line so aside fro right after the "filename = path_too_long" there are no newlines in the final registration.lta which causes the segfault. So this just needs to change to newfile.append(line+'\n') and it all seems to come out fine, or line 220 could change to Path(lta_file).write_text('\n'.join(newfile)).

**Comment by effigies** [comment]
I think you're right. I think in https://github.com/poldracklab/niworkflows/commit/e467e4df365833b17874ddaaf4245eaa32a56636, we failed to notice that `lines = Path(lta_file).read_text().splitlines()` strips the newlines. Would you care to submit a PR to niworkflows fixing the issue? We can either use `splitlines(keepends=True)` to keep the patch small, or update the rest to assume no newlines.

**Comment by davhunt** [comment]
Ah, yeah sure, thanks.

**Comment by oesteban** [comment]
Fixed via poldracklab/niworkflows#391

Please reopen if the issue regresses.
